configuration_files/trajectory_builder.lua

```diff
@@ -12,11 +12,13 @@
 -- See the License for the specific language governing permissions and
 -- limitations under the License.
 
+include "initial_trajectory_pose.lua"
 include "trajectory_builder_2d.lua"
 include "trajectory_builder_3d.lua"
 
 TRAJECTORY_BUILDER = {
   trajectory_builder_2d = TRAJECTORY_BUILDER_2D,
   trajectory_builder_3d = TRAJECTORY_BUILDER_3D,
   pure_localization = false,
+  initial_trajectory_pose = INITIAL_TRAJECTORY_POSE,
```

Comment:
I am not sure this is a good approach. While it is symmetric to how pure_localization is handled, this is quite different in the use case.

Pure localization always goes along with different settings, for example the loop closure will be run more often usually and maybe less aggressive settings for search windows. 

Initial pose however is very dependant on the data set in question: once you have one run finished, you want to start a second and relate the initial pose to this particular state. It sounds more like a flag than a configuration paramater to me. Therefore I feel that a commandline flag would be more appropriate for it.

Wdyt?

Reply:
Yes it is cumbersome to do it this way and as you mentioned it was to be consistent with pure localization. Can you elaborate a bit about command line flag? Is it a flag that should be passed to start_trajectory_main in cartographer_ros? If yes, what flag do you have in mind, something like:
```
-initial_pose "{ to_trajectory_id: 1, timestamp: now, translation : { 0, 0, 0}, rotation : { ... }}"
```